### ISSUE TITLE
Deprecation Notice on support for boto/older API's

### ISSUE THREAD

**Issue by bblommers** [spulec/moto#4526]
## Deprecation Notice for boto

As discussed in #2950, it is about time for us to drop support for Boto2. AWS has dropped support for this a long time ago, and now so will we.
For now, users can mock calls to the `boto`-module using the deprecated decorators, such as `@mock_ec2_deprecated`, but these decorators will be **removed in Moto 3.0**.

## Changes to other decorators
Two other old decorators will be changed:
 - `@mock_dynamodb`
 - `@mock_rds`

Users have long been adviced to use  the later`@mock_dynamodb2` and `@mock_rds2` instead, as those decorators are fully compatible with boto3 and support all mocked features.

The old decorators will be **repurposed in Moto 3.1**, and their functionality will become synonymous with `@mock_dynamodb2` and `@mock_rds2`.

### Changes on a per-version basis:

| Version        | Change                                                                                                                 |
|-----------------|----------------------------------------------------------------------------------------------------------|
| Moto 2.2.13 | Warnings will be added to `@mock_dynamodb` and `@mock_rds`                     |
| Moto 3.0      | Removal of all `@mock_{service}_deprecated`-decorators                                  |
| Moto 3.1      | `@mock_dynamodb` will have the same functionality as `@mock_dynamodb2` |
|                     | `@mock_rds` will have the same functionality as `@mock_rds2`                         | 
|                     |`@mock_dynamodb2` and `@mock_rds2`-decorators will be deprecated            | 
| Moto 4.0      |`@mock_dynamodb2` and `@mock_rds2`-decorators will be removed.                |

### Timeline
An exact timeline is TBD. If you are impacted by this change, we advise you to:
 - Watch this issue for further updates
 - Pin Moto to a major version (`moto < 3`), to avoid breaking things until you are ready to upgrade.

### Feedback
Please let us know below whether you have any questions/feedback on these changes.

--------
#### Notes

1) `mock_dynamodb` originally supported an older API version, 20111205.
    `mock_dynamodb2` supports the latest API version, 20120810.
    AWS still supports the older version when using the HTTP API, but only when specifying an additional header: `"X-Amz-Target": "DynamoDB_20111205.CreateTable"`.
    Moto supports this header now, and will continue to do so for the foreseeable future.

**Comment by kazanzhy** [comment]
So for now I should use `@mock_rds2` for my tests, and before version 4.0 move to `@mock_rds` back.
Is that correct?

**Comment by bblommers** [comment]
Yes, that's correct @kazanzhy. Don't worry - we'll give plenty of notices before completely removing the `@mock_rds2`-decorator.

**Comment by bblommers** [comment]
@kazanzhy Moto 3.1.0 has now been released. 

The `mock_rds2` decorator still exists with the same functionality, but it is recommended to now start switching the decorators to `mock_rds` instead.
